In [21]:
# Introduction to Model Building

from pandas import read_csv
from numpy import set_printoptions 

dataframe = read_csv('BBC.csv')

# Show first five rows of dataframe
dataframe.head()

MaritalStatus  Gender  YearlyIncome  TotalChildren  NumberChildrenAtHome  \
0              5       1         90000              2                     0   
1              5       1         60000              3                     3   
2              5       1         60000              3                     3   
3              5       2         70000              0                     0   
4              5       2         80000              5                     5   

   EnglishEducation  HouseOwnerFlag  NumberCarsOwned  CommuteDistance  Region  \
0                 5               1                0                2       2   
1                 5               0                1                1       2   
2                 5               1                1                5       2   
3                 5               0                1               10       2   
4                 5               1                4                2       2   

   Age  BikeBuyer  
0   50          1  
1   51          1  
2   51          1  
3   49          1  
4   48          1

In [12]:
# For seeing current working directory

import os

print(os.getcwd())

C:\Users\Pavan


In [13]:
# shows last five rows of dataframe
dataframe.tail(5)

MaritalStatus  Gender  YearlyIncome  TotalChildren  \
18479              4       1         30000              1   
18480              4       2         30000              3   
18481              4       1         30000              3   
18482              5       1         30000              3   
18483              5       1         30000              0   

       NumberChildrenAtHome  EnglishEducation  HouseOwnerFlag  \
18479                     0                 4               1   
18480                     0                 4               1   
18481                     0                 4               0   
18482                     0                 5               1   
18483                     0                 5               1   

       NumberCarsOwned  CommuteDistance  Region  Age  BikeBuyer  
18479                0                1       3   58          1  
18480                0                1       3   56          1  
18481                0                1       3   57          1  
18482                0                1       3   58          1  
18483                0                1       3   57          1

In [14]:
# shows shape(rows, columns) of a dataframe
dataframe.shape

(18484, 12)

In [15]:
# shows mean, median, mode of the data

# mean = average
# median = values in the middle of the rage
# mode = The number which appears most often in a set of numbers

dataframe.describe()

MaritalStatus        Gender   YearlyIncome  TotalChildren  \
count   18484.000000  18484.000000   18484.000000   18484.000000   
mean        3.310106      1.494103   57305.777970       1.844352   
std         1.426147      0.499979   32285.841703       1.612408   
min         1.000000      1.000000   10000.000000       0.000000   
25%         2.000000      1.000000   30000.000000       0.000000   
50%         3.000000      1.000000   60000.000000       2.000000   
75%         5.000000      2.000000   70000.000000       3.000000   
max         5.000000      2.000000  170000.000000       5.000000   

       NumberChildrenAtHome  EnglishEducation  HouseOwnerFlag  \
count          18484.000000      18484.000000    18484.000000   
mean               1.004058          3.310106        0.676369   
std                1.522660          1.426147        0.467874   
min                0.000000          1.000000        0.000000   
25%                0.000000          2.000000        0.000000   
50%                0.000000          3.000000        1.000000   
75%                2.000000          5.000000        1.000000   
max                5.000000          5.000000        1.000000   

       NumberCarsOwned  CommuteDistance        Region           Age  \
count     18484.000000     18484.000000  18484.000000  18484.000000   
mean          1.502705         6.003246      1.789710     54.603928   
std           1.138394         6.382562      0.872664     11.255252   
min           0.000000         1.000000      1.000000     36.000000   
25%           1.000000         1.000000      1.000000     46.000000   
50%           2.000000         2.000000      1.000000     53.000000   
75%           2.000000        10.000000      3.000000     62.000000   
max           4.000000        20.000000      3.000000    106.000000   

          BikeBuyer  
count  18484.000000  
mean       0.494049  
std        0.499978  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

In [16]:
dataframe.values

array([[    5,     1, 90000, ...,     2,    50,     1],
       [    5,     1, 60000, ...,     2,    51,     1],
       [    5,     1, 60000, ...,     2,    51,     1],
       ...,
       [    4,     1, 30000, ...,     3,    57,     1],
       [    5,     1, 30000, ...,     3,    58,     1],
       [    5,     1, 30000, ...,     3,    57,     1]], dtype=int64)

In [18]:
# include all items in the dataframe from 0 to 11
dataframe.values[:,0:11]

array([[    5,     1, 90000, ...,     2,     2,    50],
       [    5,     1, 60000, ...,     1,     2,    51],
       [    5,     1, 60000, ...,     5,     2,    51],
       ...,
       [    4,     1, 30000, ...,     1,     3,    57],
       [    5,     1, 30000, ...,     1,     3,    58],
       [    5,     1, 30000, ...,     1,     3,    57]], dtype=int64)

In [40]:
# column in the dataframe mapped to the 11th row
dataframe.values[:, 11]

array([1, 1, 1, ..., 1, 1, 1], dtype=int64)

In [33]:
# shows only bikebuyer column
bike_buyers = dataframe.groupby('BikeBuyer')

# size shows number of distributions of o's(non_bike_buyers) and 1'(bike_buyers)
bike_buyers.size()

BikeBuyer
0    9352
1    9132
dtype: int64

Number of bikers and non-bike buyers are evenly distributed
As we can see the distributions 9352(non_bike_buyers) and 9152(bike_buyers) above.

Even number of distributions really matters for finding the accuracy of our predictive model.




# Creation of model to check the accuracy


[Building model using SVC]

In [52]:
from pandas import read_csv
from sklearn.model_selection import train_test_split     # For testing
from sklearn.svm import SVC                              # Classifier

dataframe = read_csv('BBC.csv')

x = dataframe.values[:, 0:11]                           # Whole Training data, Includes all items in the dataframe from 0 to 11
y = dataframe.values[:, 11]                             # BikeBuyers column in the dataframe mapped to the 11th row

# In the cell below we are specifying the test size. 
# If our test size is 25% of the data then our training size is 75% of the data.
# The sole pupose of seed is to reproduce the results.

test_size = 0.30
seed = 45

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = test_size, random_state = seed)

model = SVC(gamma = 'auto')                       # Setting gamma explicitly to auto for the purpose of avoiding future warning
model.fit(x_test, y_test)

result = model.score(x_test, y_test) * 100
print('Accuracy is {}'.format(result))


Accuracy is 81.87883159033538



[Building model using KNeighborsClassifier]

In [53]:
from pandas import read_csv
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier

dataframe = read_csv('BBC.csv')

x = dataframe.values[:, 0:11]                           # Whole Training data, Includes all items in the dataframe from 0 to 11
y = dataframe.values[:, 11]                             # BikeBuyers column in the dataframe mapped to the 11th row

# In the cell below we are specifying the test size. 
# If our test size is 25% of the data then our training size is 75% of the data.
# The sole pupose of seed is to reproduce the results.

test_size = 0.30
seed = 45

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = test_size, random_state = seed)

model = KNeighborsClassifier()                    
model.fit(x_test, y_test)

result = model.score(x_test, y_test) * 100
print('Accuracy is {}'.format(result))

Accuracy is 82.58204111071042



[Building Model Using RandomForestClassifier]

In [55]:
from pandas import read_csv
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier

dataframe = read_csv('BBC.csv')

x = dataframe.values[:, 0:11]                           # Whole Training data, Includes all items in the dataframe from 0 to 11
y = dataframe.values[:, 11]                             # BikeBuyers column in the dataframe mapped to the 11th row

# In the cell below we are specifying the test size. 
# If our test size is 25% of the data then our training size is 75% of the data.
# The sole pupose of seed is to reproduce the results.

test_size = 0.30
seed = 45

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = test_size, random_state = seed)

model = RandomForestClassifier(n_estimators = 100)      # Setting n_estimator to 100 for the purpose of avoiding future warning
model.fit(x_test, y_test)

result = model.score(x_test, y_test) * 100
print('Accuracy is {}'.format(result))

Accuracy is 90.75009015506672
